src/skin/qml/qmlplayermanagerproxy.cpp

```diff
@@ -17,7 +20,9 @@ QObject* QmlPlayerManagerProxy::getPlayer(const QString& group) {
         qWarning() << "PlayerManagerProxy failed to find player for group" << group;
         return nullptr;
     }
-    return new QmlPlayerProxy(pPlayer, this);
+    QmlPlayerProxy* pPlayerProxy = new QmlPlayerProxy(pPlayer, this);
```

Comment:
The `this` argument is confusing if you reparent the newly created object immediately.

Reply:
Why is QmlPlayerManagerProxy not suitable as the owner?

Reply:
I guess I understand why. The manager will create many independent instances because we don't cache them by group. Please add a comment.

Reply:
> Why is QmlPlayerManagerProxy not suitable as the owner?

The QmlPlayerManagerProxy is a *singleton* and will live for the whole lifetime of the QML engine. But I can create a bunch of PlayerManager instances in a JavaScript loop, and then drop all references to them. If we don't explicitly set JS ownership and just rely on the QmlPlayerManagerProxy parentship, then these instances will live as long as the parent instead of being deleted when the last JS reference to it was dropped.

Reply:
Your comment made me revisit the documentation. The object mustn't have a parent :D